// @airgram/core/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !exports.hasOwnProperty(p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TDLibError = exports.Provider = exports.Composer = exports.AirgramCore = void 0;
var AirgramCore_1 = require("./AirgramCore");
Object.defineProperty(exports, "AirgramCore", { enumerable: true, get: function () { return AirgramCore_1.AirgramCore; } });
var components_1 = require("./components");
Object.defineProperty(exports, "Composer", { enumerable: true, get: function () { return components_1.Composer; } });
Object.defineProperty(exports, "TDLibError", { enumerable: true, get: function () { return components_1.TDLibError; } });
Object.defineProperty(exports, "Provider", { enumerable: true, get: function () { return components_1.Provider; } });
__exportStar(require("./types"), exports);
var utils_1 = require("./utils");
Object.defineProperty(exports, "isError", { enumerable: true, get: function () { return utils_1.isError; } });
Object.defineProperty(exports, "toObject", { enumerable: true, get: function () { return utils_1.toObject; } });
Object.defineProperty(exports, "on", { enumerable: true, get: function () { return utils_1.on; } });
Object.defineProperty(exports, "createDeferred", { enumerable: true, get: function () { return utils_1.createDeferred; } });
