// airgram/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !exports.hasOwnProperty(p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.prompt = exports.createSerializer = exports.createDeserializer = exports.TdJsonClient = exports.Auth = exports.Airgram = void 0;
var Airgram_1 = require("./Airgram");
Object.defineProperty(exports, "Airgram", { enumerable: true, get: function () { return Airgram_1.Airgram; } });
var components_1 = require("./components");
Object.defineProperty(exports, "Auth", { enumerable: true, get: function () { return components_1.Auth; } });
Object.defineProperty(exports, "TdJsonClient", { enumerable: true, get: function () { return components_1.TdJsonClient; } });
var utils_1 = require("./utils");
Object.defineProperty(exports, "createDeserializer", { enumerable: true, get: function () { return utils_1.createDeserializer; } });
Object.defineProperty(exports, "createSerializer", { enumerable: true, get: function () { return utils_1.createSerializer; } });
Object.defineProperty(exports, "prompt", { enumerable: true, get: function () { return utils_1.prompt; } });
__exportStar(require("@airgram/core"), exports);
